radicle-httpd/src/lib.rs

```diff
@@ -79,3 +100,58 @@ pub async fn run(options: Options) -> anyhow::Result<()> {
         .await
         .map_err(anyhow::Error::from)
 }
+
+#[derive(Clone)]
+struct IdCounter(Arc<AtomicU64>);
+
+impl IdCounter {
+    fn new() -> IdCounter {
+        IdCounter(Arc::new(0.into()))
+    }
+
+    fn next(&mut self) -> u64 {
+        self.0.fetch_add(1, Ordering::SeqCst)
+    }
+}
+
+struct TracingInfo {
+    connect_info: ConnectInfo<SocketAddr>,
+    method: Method,
+    version: Version,
+    uri: Uri,
+}
+
+struct ColoredStatus(StatusCode);
+
+impl fmt::Display for ColoredStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.0.as_u16() {
+            200..=299 => write!(f, "{}", Paint::green(self.0)),
+            300..=399 => write!(f, "{}", Paint::blue(self.0)),
+            400..=499 => write!(f, "{}", Paint::red(self.0)),
+            _ => write!(f, "{}", Paint::yellow(self.0)),
```

Comment:
something like [this](https://www.infidigit.com/wp-content/uploads/2019/12/20191227_012601_0000.png)